app/src/main/java/app/revanced/integrations/sponsorblock/player/ui/SponsorBlockView.java

```diff
@@ -109,8 +116,13 @@ private static void skipSponsorButtonVisibility(boolean visible) {
 
         visible &= shouldShowOnPlayerType;
 
-        skipSponsorButton.setVisibility(visible ? View.VISIBLE : View.GONE);
-        bringLayoutToFront();
+        final int desiredVisibility = visible ? View.VISIBLE : View.GONE;
+        if (skipSponsorButton.getVisibility() != desiredVisibility) {
+            skipSponsorButton.setVisibility(desiredVisibility);
+            if (visible) {
+                bringLayoutToFront();
+            }
+        }
```

Comment:
Is this logic necessary? The Android api already checks if the visibility has changed:

```java
    void setFlags(int flags, int mask) {
        // ....

        int old = mViewFlags;
        mViewFlags = (mViewFlags & ~mask) | (flags & mask);

        int changed = mViewFlags ^ old;
        if (changed == 0) {
            return;
        }
        
        // ....
}
```